api/views/workflow.py

```diff
@@ -35,7 +35,7 @@ class SourceViewSet(viewsets.ReadOnlyModelViewSet):
     serializer_class = SourceSerializer
 
     def get_queryset(self):
-        queryset = Source.objects.all()
+        queryset = Source.objects.exclude(icon='')
```

Comment:
When clearing an image file through the admin interface, the field isn't nulled. I can't find any documentation on this though?